apps/essimporter/converter.cpp

```diff
@@ -0,0 +1,241 @@
+            esm.skipHSub(); // skip CNDT
+        }
+
+        std::vector<CellRef> cellrefs;
+        while (esm.hasMoreSubs() && esm.isNextSub("FRMR"))
+        {
+            CellRef ref;
+            ref.load (esm);
+            if (esm.isNextSub("DELE"))
+            {
+                // strangely this can be e.g. 52 instead of just 1,
+                std::cout << "deleted ref " << ref.mIndexedRefId << std::endl;
+                esm.skipHSub();
+            }
+            cellrefs.push_back(ref);
+        }
+
+        while (esm.isNextSub("MPCD"))
+        {
+            float notepos[3];
+            esm.getHT(notepos, 3*sizeof(float));
+
+            // Markers seem to be arranged in a 32*32 grid, notepos has grid-indices.
+            // This seems to be the reason markers can't be placed everywhere in interior cells,
+            // i.e. when the grid is exceeded.
+            // Converting the interior markers correctly could be rather tricky, but is probably similar logic
+            // as used for the FoW texture placement, which we need to figure out anyway
+            notepos[1] += 31.f;
+            notepos[0] += 0.5;
+            notepos[1] += 0.5;
+            notepos[0] = 8192 * notepos[0] / 32.f;
+            notepos[1] = 8192 * notepos[1] / 32.f;
+            if (cell.isExterior())
+            {
+                notepos[0] += 8192 * cell.mData.mX;
+                notepos[1] += 8192 * cell.mData.mY;
+            }
+            // TODO: what encoding is this in?
+            std::string note = esm.getHNString("MPNT");
+            ESM::CustomMarker marker;
+            marker.mWorldX = notepos[0];
+            marker.mWorldY = notepos[1];
+            marker.mNote = note;
+            marker.mCell = cell.getCellId();
+            mMarkers.push_back(marker);
+        }
+
+        newcell.mRefs = cellrefs;
+
+        // FIXME: map by ID for exterior cells
+        mCells[id] = newcell;
+    }
+
+    void ConvertCell::write(ESM::ESMWriter &esm)
+    {
+        for (std::map<std::string, Cell>::const_iterator it = mCells.begin(); it != mCells.end(); ++it)
+        {
+            const ESM::Cell& cell = it->second.mCell;
+            esm.startRecord(ESM::REC_CSTA);
+            ESM::CellState csta;
+            csta.mHasFogOfWar = 0;
+            csta.mId = cell.getCellId();
+            csta.mId.save(esm);
+            // TODO csta.mLastRespawn;
+            // shouldn't be needed if we respawn on global schedule like in original MW
+            csta.mWaterLevel = cell.mWater;
+            csta.save(esm);
+
+            for (std::vector<CellRef>::const_iterator refIt = it->second.mRefs.begin(); refIt != it->second.mRefs.end(); ++refIt)
+            {
+                const CellRef& cellref = *refIt;
+                ESM::CellRef out;
+                out.blank();
+
+                if (cellref.mIndexedRefId.size() < 8)
+                {
+                    std::cerr << "CellRef with no index?" << std::endl;
```

Comment:
This message is no longer relevant as of now (code I haven't pushed yet, part of a large WIP commit), so I will not fix it.
In short, CellRefs with no index are actually very common.